src/openms/include/OpenMS/CHEMISTRY/ElementDB.h

```diff
@@ -127,30 +166,42 @@ namespace OpenMS
 
 protected:
 
-    /** parses a isotope distribution of abundances and masses
-
-    **/
-    IsotopeDistribution parseIsotopeDistribution_(const std::map<unsigned int, double>& abundance, const std::map<unsigned int, double>& mass);
-
     /** calculates the average weight based on isotope abundance and mass
      **/
     double calculateAvgWeight_(const std::map<unsigned int, double>& abundance, const std::map<unsigned int, double>& mass);
 
-    /**_ calculates the mono weight based on the most abundant isotope 
+    /**_ calculates the mono weight based on the most abundant isotope
      **/
     double calculateMonoWeight_(const std::map<unsigned int, double>& abundance, const std::map<unsigned int, double>& mass);
 
 	  /// constructs element objects
     void storeElements_();
 
-    /// build element objects from given abundances, masses, name, symbol, and atomic number
-    void buildElement_(const std::string& name, const std::string& symbol, const unsigned int an, const std::map<unsigned int, double>& abundance, const std::map<unsigned int, double>& mass);
+    /// build element objects from given abundances, masses, half lifes, decay mode name, symbol, and atomic number
+    void buildElement_(const std::string& name,
+                       const std::string& symbol,
+                       const unsigned int an,
+                       const std::map<unsigned int, double>& abundance,
+                       const std::map<unsigned int, double>& mass,
+                       const std::map<unsigned int, double>& half_lifes =
+                         std::map<unsigned int, double>(),
+                       const std::map<unsigned int, Isotope::DecayMode>& decay_modes =
+                         std::map<unsigned int, Isotope::DecayMode>());
+
+    /// helper to build all isotopes from an input list
+    std::vector<const Isotope *> buildIsotopes_(const std::string& name,
+                        const std::string& symbol,
+                        const unsigned int an,
+                        const std::map<unsigned int, double>& abundance,
+                        const std::map<unsigned int, double>& mass,
+                        const std::map<unsigned int, double>& half_life,
+                        const std::map<unsigned int, Isotope::DecayMode>& decay);
 
     /// add element objects to documentation maps
-    void addElementToMaps_(const std::string& name, const std::string& symbol, const unsigned int an, const Element* e);
+    Element* addElementToMaps_(const std::string& name, const std::string& symbol, const unsigned int an, Element* e);
```

Comment:
```suggestion
    const Element* addElementToMaps_(const std::string& name, const std::string& symbol, const unsigned int an, const Element* e);
```

Reply:
this can be made `const`, if I'm not mistaken.

Reply:
`addElementToMaps_` cannot but `addIsotopeToMaps_` can, the reason is that we need to call a non-const function on the Element in order to update the isotopes

Reply:
Sorry, I still don't see it...
If you write `e = const_ele;` here https://github.com/OpenMS/OpenMS/pull/5956/files#diff-8ebe022fac5a9f3c0eb06f940c6706d8d9bb652131218ee474d495451af8627dR707, why would that not work?